### PR TITLE
TBE-71 JMeter tests

### PR DESCRIPTION
**Pull request by tofarr**
## Is PM acceptance required?
- [ ] Yes - don't merge until JIRA issue is accepted!
- [x] No - merge after code review approval
## What was done?
- Created JMeter Tests and bash scripts for vita-min
- Test scripts try to mimic real usage as found in datadog
- Tests may be run against dev or staging
## How to test?
- See the README.md for details on Jmeter setup
- Run the command `jmeter_test/jmeter_test -t fyst_az_5_minute_stress_test`
### Local Results from my Laptop
image
[TBE-71]: https://codeforamerica.atlassian.net/browse/TBE-71?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: jmeter_test/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * `jmeter_test/jmeter_test -t fyst_az_5_minute_stress_test -g` : Open `fyst_az_5_minute_stress_test` plan in GUI
+
+
+## Running in AWS
+
+This will create a docker image that can be deployed to Fargate / ECS using serverless.
+
+ * Download docker from https://docker.com
+ * From `jmeter_tests/` Create an image with `docker build -t jmeter_test .`
+ * Make sure your local AWS settings are in place.
```
Comment (jenny-heath) [inline jmeter_test/README.md:33]: i'm not sure what settings this refers to, can this be more specific?

File: jmeter_test/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ * Deploy with `sls deploy`
```
Comment (jenny-heath) [inline jmeter_test/README.md:34]: is this run from the same directory as step 2 or somewhere else?

Comment (tofarr) [reply jmeter_test/README.md]: Updated the Readme with more details. I'm not gonna use TFM for now

File: jmeter_test/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,41 @@
+FROM alpine:3.10
+MAINTAINER tofarr tofarrell@codeforamerica.org
+
+# Install packages
+#RUN apk update && \
+#    apk add ca-certificates wget python python-dev py-pip && \
+#    update-ca-certificates && \
+#    pip install --upgrade --user awscli
```
Comment (jenny-heath) [inline jmeter_test/Dockerfile:8]: can we delete this since it's commented out?

File: jmeter_test/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+COPY fyst_az_5_minute_stress_test.jmx $TESTDIR
+COPY entrypoint.sh $TESTDIR
+COPY jmeter_test $TESTDIR
+RUN mkdir -p ${TESTDIR}results
+
+# Not working
+#ENTRYPOINT ["./jmeter_test.sh -t fyst_az_5_minute_stress_test -e staging"]
+
+# Working
+#ENTRYPOINT ["ls"]
```
Comment (jenny-heath) [inline jmeter_test/Dockerfile:39]: ^ same

File: jmeter_test/README.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+ * `jmeter_test/jmeter_test -t fyst_az_5_minute_stress_test` : Run `fyst_az_5_minute_stress_test` plan against dev
+ * `jmeter_test/jmeter_test -t fyst_az_5_minute_stress_test -e staging` : Run `fyst_az_5_minute_stress_test` plan against staging
+ * `jmeter_test/jmeter_test -t fyst_az_5_minute_stress_test -g` : Open `fyst_az_5_minute_stress_test` plan in GUI
+
+
+## Running in AWS
+
+Running JMeter from a developer machine is possible, though 
+This will create a docker image that can be deployed to Fargate / ECS.
```
Comment (jenny-heath) [inline jmeter_test/README.md:30]: more clear, maybe:

```suggestion
While running JMeter from a local developer machine is possible (and necessary for developing test plans), testing production-like environments must be done within AWS to not trigger anti-DDOS protections.

Use these instructions to create a docker image that can be deployed to Fargate / ECS for testing production-like environments.
```

File: jmeter_test/README.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+ * Download docker from https://docker.com
+ * Cd into the `jmeter_tests/` directory, and create an image with `docker build -t jmeter_test .`
+ * Make sure your local AWS config (in $HOME/.aws) contains your credentials for AWS.
+ * Create a Repository in Amazon Elastic Container Registry (ECR). I created `tim-jmeter-test` in https://us-east-1.console.aws.amazon.com/ecr/private-registry/repositories?region=us-east-1
+ * Push your docker image to your Repository. (Click "View Push commands" at https://us-east-1.console.aws.amazon.com/ecr/private-registry/repositories?region=us-east-1)
+ * Create a fargate cluster at https://us-east-1.console.aws.amazon.com/ecs/v2/clusters?region=us-east-1 (I created `tim-jmeter-test`)
+ * Create a task to run your docker image on your cluster. For "Compute Options", I chose "Launch Type" with "FARGATE".
+   My application type was "Task", and "jmeter-test" was the "Family". For "Networking", I chose the "tax-benefits-dashboards-vpc" with all subnets.
+   I chose the "default" and "VPN Security Group" Security Groups. I enabled a public IP.
```
Comment (jenny-heath) [inline jmeter_test/README.md:40]: would be good to include a line about how to get the test results out of the Fargate container

Comment (jenny-heath) [reply jmeter_test/README.md:40]: oh and also how to tear it down after you're done

**Review by jnf — CHANGES_REQUESTED** [review]
hi hi! reviewed with Jenny and have a couple minor requests! mostly documentation-related. 😅

File: jmeter_test/Dockerfile (hunk 3 above)
Comment (tofarr) [reply jmeter_test/Dockerfile]: Done!

File: jmeter_test/Dockerfile (hunk 4 above)
Comment (tofarr) [reply jmeter_test/Dockerfile]: Done!

**Review by jenny-heath — APPROVED** [review]

**Review by jnf — APPROVED** [review]

**Pull request merged**